File: 백준/Silver/10828. 스택/스택.c
#include <stdio.h>
#include <string.h>

int stack[10001];
int top = -1;

void push(int data){
    stack[++top] = data;
}
int pop(){
    return stack[top--];
}

int main(void){
    int N, X;
    char op[6];
    scanf("%d", &N);
    while(N--){
        scanf("%s", op);
        if(op[0] == 'p'){ // push, pop
            if(op[1] == 'u'){
                scanf("%d", &X);
                push(X);
            }else{
                if(top == -1) printf("-1\n");
                else printf("%d\n", pop()); 
            }
        }else if(op[0] == 't'){ // top
            if(top == -1) printf("-1\n");
            else printf("%d\n", stack[top]);
        }else if(op[0] == 's'){ // size
            printf("%d\n", top + 1);
        }else if(op[0] == 'e'){ // empty
            if(top == -1) printf("1\n");
            else printf("0\n");
        }
    }
}

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {
    public void solution() throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        Stack<Integer> stack = new Stack<>();

        int N = Integer.parseInt(br.readLine());
        for(int i = 0; i < N; i++){
            StringTokenizer st = new StringTokenizer(br.readLine());
            String str = st.nextToken();
            
            if("push".equals(str)){
                int data = Integer.parseInt(st.nextToken());
                stack.push(data);
            }else if("pop".equals(str)){
                if(stack.size() == 0) bw.write("-1\n"); 
                else{
                    int data = stack.pop();
                    bw.write(String.valueOf(data) + '\n');
                }
            }else if("size".equals(str)){
                int size = stack.size();
                bw.write(String.valueOf(size) + '\n'); 
            }else if("empty".equals(str)){
                boolean isEmpty = stack.isEmpty();
                if(isEmpty) bw.write("1\n");
                else bw.write("0\n");
            }else{
                if(stack.size() == 0) bw.write("-1\n"); 
                else{
                    int data = stack.peek();
                    bw.write(String.valueOf(data) + '\n');
                }
            }
        }
        bw.flush();
        bw.close();
        br.close();
    }
    public static void main(String[] args) throws Exception {
        new Main().solution();
    }
}
